signac/core/h5store.py

```diff
@@ -257,11 +257,25 @@ class H5Store(MutableMapping):
             assert h5s.foo == 'bar'
             assert h5s['foo'] == 'bar'
 
+    The H5Store can be used as a context manager to ensure that the underlying file
+    is opened, however most built-in types can be read and stored without the need
+    to _explicitly_ open the file. However, to access arrays (reading or writing),
+    the file must always be opened!
+
+    To open a file in read-only mode, use the :py:meth:`.open` method with ``mode=r``:
+
+    .. code-block:: python
+
+        with H5Store('fileh5').open(mode='r') as h5s:
+            pass
+
     :param filename:
         The filename of the underlying HDF5-file.
     :param kwargs:
-        Additional keyword arguments to be forward to the h5py.File constructor
-        See documentation for :class:`h5py.File` for more information.
+        Additional keyword arguments to be forward to the ``h5py.File`` constructor
+        See the documentation for the
+        `h5py.File constructor <http://docs.h5py.org/en/latest/high/file.html#File>`_
```

Comment:
Should this link use intersphinx instead?

Reply:
I tried, doesn't look they expose it properly. We can possibly fix that in a subsequent release.